entoas/extension.go

```diff
@@ -0,0 +1,183 @@
+package entoas
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+
+	"entgo.io/contrib/entoas/policy"
+	"entgo.io/contrib/entoas/spec"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+)
+
+type (
+	// Config provides global metadata for the generator. Is injected into the *gen.Graph.
+	Config struct {
+		// DefaultPolicy defines the default policy for endpoint generation.
+		// It is used if no policy is set on a (sub-)resource.
+		// Defaults to PolicyExpose.
+		DefaultPolicy policy.Policy
+		// Whether or whether not to generate simple models instead of one model per endpoint.
+		SimpleModels bool
+	}
+	// Extension implements entc.Extension interface for providing http handler code generation.
+	Extension struct {
+		entc.DefaultExtension
+		config    *Config
+		mutations []MutateFunc
+		out       io.Writer
+	}
+	// ExtensionOption allows managing Extension configuration using functional arguments.
+	ExtensionOption func(*Extension) error
+)
+
+// NewExtension returns a new elk extension with default values.
```

Comment:
Who is elk? 🙃

While your at it feel the project to find other usages

Reply:
> While your at it feel the project to find other usages

Ahm, I am not sure I understand :sweat_smile: